server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -86,9 +90,18 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
             return subQuery.createWeight(searcher, scoreMode, boost);
         }
         boolean needsScore = scriptBuilder.needs_score();
-        ScoreMode subQueryScoreMode = needsScore ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
+        boolean needsTermStatistics = scriptBuilder.needs_termStatistics();
+
```

Comment:
ℹ️ Required so termStats are collected during DFS when `needs_termStatistics` is true.